dart/dynamics/FreeJoint.cpp

```diff
@@ -575,6 +556,16 @@ const std::string& FreeJoint::getStaticType()
 }
 
 //==============================================================================
+void FreeJoint::setTransformFromChildBodyNode(const Eigen::Isometry3d& T)
+{
+  Joint::setTransformFromChildBodyNode(T);
+
+  // We update Jacobian here since the Jacobian of FreeJoint is a function of
+  // "T" but not of joint positions.
+  mJacobian = math::getAdTMatrix(T);
```

Comment:
I think we should move this to ``updateLocalJacobian()`` and call ``updateLocalJacobian()`` here. Then in the ``updateLocalJacobian()`` function we would wrap this up in the ``_mandatory`` flag so the computation only gets performed when something demands it.